File: app/src/main/java/com/argos/android/opencv/interfaces/MainActivityCallback.java
package com.argos.android.opencv.interfaces;

import com.argos.android.opencv.model.Feature;

/**
 * Interfacing {@link com.argos.android.opencv.adapter.FeatureListAdapter} -> {@link com.argos.android.opencv.activity.MainActivity}
 */

public interface MainActivityCallback
{
    void chooseImage(Feature feature);
    void launchCameraDetection(Feature feature);
}

File: app/src/main/java/com/argos/android/opencv/interfaces/MainActivityCallback.kt
package com.argos.android.opencv.interfaces

import com.argos.android.opencv.model.Feature

/**
 * Interfacing [com.argos.android.opencv.adapter.FeatureListAdapter] -> [com.argos.android.opencv.activity.MainActivity]
 */

interface MainActivityCallback {
    fun chooseImage(feature: Feature)
    fun launchCameraDetection(feature: Feature)
}
